### ISSUE TITLE
Indicate memory requirements per-model

### ISSUE THREAD

**Issue by leoll2** [open-edge-platform/training_extensions#7379]
## Objective

Help users pick the right model to fine-tune based on their available memory (RAM, GPU) and the requirements of each model architecture.

## Motivation

Geti offers 60+ models across different tasks. Some model architectures are lightweight and designed for speed at the edge, while the bigger models are more focused on achieving higher accuracy and are consequently more demanding to train in terms of HW resources. As of v3.1, Geti shows the following information to characterize the models:
- Number of parameters
- Gigaflops
- Tag "accuracy" / "speed" / "balance" (only for some models)

While these properties are certainly useful to know, they don't directly address one fundamental user question: 
> Does the model fit in my machine?

The answer is not straightforward, since the memory requirement doesn't only depend on the model architecture but also on the batch size, number of categories and other parameters. Nevertheless, it would be helpful to provide a rough indication of the minimum HW requirements needed for each model, so that the user can take advantage of this information to decide what model to pick. He can still attempt to train a model architecture whose requirements are beyond the available resources, in that case the user is aware that training may fail.

## Requirements

- Identify the minimum memory required for each model architecture.
- Publish the minimum requirements in the documentation and, possibly, in the UI too.
- Monitor changes in the requirements over time, by means of automated CI workflows that periodically profile each algorithm.
